packages/pixeloven/cli/src/commands/init.ts

```diff
@@ -0,0 +1,131 @@
+import { getKeys } from "@pixeloven-core/common";
+import * as Validation from "@pixeloven-core/validation";
+import { PixelOvenToolbox } from "../types";
+
+enum ProjectType {
+    New,
+    Existing,
+}
+type ProjectStrings = keyof typeof ProjectType;
+
+enum PackageManager {
+    NPM,
+    Yarn,
+}
+type PackageManagerStrings = keyof typeof PackageManager;
+
+interface ProjectOptions {
+    projectType: ProjectStrings;
+}
+
+interface NewProjectOptions {
+    projectName: string;
+    projectAuthor: string;
+    projectDescription: string;
+    projectPackageManager: PackageManagerStrings;
+}
+
+export default {
+    description: "PixelOven initializer",
+    name: "init",
+    run: async (toolbox: PixelOvenToolbox) => {
+        const { prompt, print, template } = toolbox;
+
+        /**
+         * @todo init should just create the basic project and then the generator addon is installed and can be run by the user to create apps and packages
+         * @todo why does this not actually generate anything :(
+         * @todo need to make sure all paths are lower case
+         * @todo let's simplify tsconfig
+         * @todo Need to add all of our addon stuff
+         * @todo why does it assume /bu
+         * @param dist/lib/build/templates/ ????????????????????????????????????
+         */
+        async function generate(props: NewProjectOptions) {
+            const { projectName } = props;
+            await template.generate({
+                props,
+                target: `${projectName.toLowerCase()}/apps/.gitkeep`,
+                template: "project/gitkeep.ejs",
+            });
+            await template.generate({
+                props,
+                target: `${projectName.toLowerCase()}/packages/.gitkeep`,
+                template: "project/gitkeep.ejs",
+            });
+            await template.generate({
+                props,
+                target: `${projectName.toLowerCase()}/lerna.json`,
+                template: "project/lerna.json.ejs",
+            });
+            await template.generate({
+                props,
+                target: `${projectName.toLowerCase()}/package.json`,
+                template: "project/package.json.ejs",
+            });
+            await template.generate({
+                props,
+                target: `${projectName.toLowerCase()}/tsconfig.json`,
+                template: "project/tsconfig.json.ejs",
+            });
+        }
+
+        const statusCode = 0;
+        const askProjectQuestions = [
+            {
+                choices: getKeys(ProjectType),
+                message: "Is this a new or existing project?",
+                name: "projectType",
+                type: "select",
+            },
+        ];
+        const askNewProjectQuestions = [
+            {
+                choices: getKeys(PackageManager),
+                message: `Which package manager do you prefer?`,
```

Comment:
Is Prettier not able to decide if a template literal isn't necessary?

Reply:
Nah I don't think they do that yet